docs/standard/serialization/system-text-json/polymorphism.md

```diff
@@ -136,7 +139,607 @@ The following example shows the JSON that results from the preceding code:
+    }")
+
+Console.WriteLine(value is WeatherForecastWithCity) // False
+```
+
+## Polymorphic type discriminators
+
+To enable polymorphic deserialization, users need to specify a type discriminator for the derived class:
+
+```csharp
+[JsonDerivedType(typeof(WeatherForecastBase), typeDiscriminator: "base")]
+[JsonDerivedType(typeof(WeatherForecastWithCity), typeDiscriminator: "withCity")]
+public class WeatherForecastBase
+{
+    public DateTimeOffset Date { get; set; }
+    public int TemperatureCelsius { get; set; }
+    public string? Summary { get; set; }
+}
+
+public class WeatherForecastWithCity : WeatherForecastBase
+{
+    public string? City { get; set; }
+}
+```
+
+```vb
+<JsonDerivedType(GetType(WeatherForecastBase), "base")>
+<JsonDerivedType(GetType(WeatherForecastWithCity), "withCity")>
+Public Class WeatherForecastBase
+    Public Property [Date] As DateTimeOffset
+    Public Property TemperatureCelsius As Integer
+    Public Property Summary As String
+End Class
+
+Public Class WeatherForecastWithCity
+    Inherits WeatherForecastBase
+    Public Property City As String
+End Class
+```
+
+With the added metadata, specifically, the type discriminator, the serializer can serialize and deserialize the payload as the `WeatherForecastWithCity` type from its base type `WeatherForecastBase`. This will now emit JSON along with type discriminator metadata:
+
+```csharp
+WeatherForecastBase weather = new WeatherForecastWithCity
+{
+    City = "Milwaukee",
+    Date = new DateTimeOffset(2022, 9, 26, 0, 0, 0, TimeSpan.FromHours(-5)),
+    TemperatureCelsius = 15,
+    Summary = "Cool"
+}
+var json = JsonSerializer.Serialize<WeatherForecastBase>(weather, options);
+Console.WriteLine(json);
+// Sample output:
+//   {
+//     "$type" : "withCity",
+//     "City": "Milwaukee",
+//     "Date": "2022-09-26T00:00:00-05:00",
+//     "TemperatureCelsius": 15,
+//     "Summary": "Cool"
+//   }
+```
+
+```vb
+Dim weather As WeatherForecastBase = New WeatherForecastWithCity With
+{
+    .City = "Milwaukee",
+    .[Date] = New DateTimeOffset(2022, 9, 26, 0, 0, 0, TimeSpan.FromHours(-5)),
+    .TemperatureCelsius = 15,
+    .Summary = "Cool"
+}
+Dim json As String = JsonSerializer.Serialize(weather, options)
+Console.WriteLine(json)
+' Sample output:
+'   {
+'     "$type" : "withCity",
+'     "City": "Milwaukee",
+'     "Date": "2022-09-26T00:00:00-05:00",
+'     "TemperatureCelsius": 15,
+'     "Summary": "Cool"
+'   }
+```
+
+With the type discriminator, the serializer can deserialize the payload polymorphically as `WeatherForecastWithCity`:
+
+```csharp
+WeatherForecastBase value = JsonSerializer.Deserialize<WeatherForecastBase>(json);
+Console.WriteLine(value is WeatherForecastWithCity); // True
+```
+
+```vb
+Dim value As WeatherForecastBase = JsonSerializer.Deserialize(json)
+Console.WriteLine(value is WeatherForecastWithCity) // True
+```
+
+Type discriminator identifiers are valid in either `string` or `int` forms, so the following is valid:
```

Comment:
Consider adding this to the next section and renaming it to just "Type discriminator formats" (or type discriminator configurations if you prefer).

Reply:
Adjusted per recommendation, thanks.